cmd/ncproxy/main.go

```diff
@@ -35,97 +28,7 @@ var (
-	if hook, err := etwlogrus.NewHook("Microsoft.Virtualization.NCProxy"); err == nil {
-		logrus.AddHook(hook)
-	} else {
-		logrus.Error(err)
-	}
-
-	// Register our OpenCensus logrus exporter
-	trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
-	trace.RegisterExporter(&oc.LogrusExporter{})
-
-	flag.Parse()
-	ctx := context.Background()
-	conf, err := loadConfig(*configPath)
-	if err != nil {
-		log.G(ctx).WithError(err).Fatal("failed getting configuration file")
-	}
-
-	if conf.GRPCAddr == "" {
-		log.G(ctx).Fatal("missing GRPC endpoint in config")
-	}
-
-	if conf.TTRPCAddr == "" {
-		log.G(ctx).Fatal("missing TTRPC endpoint in config")
-	}
-
-	// If there's a node network service in the config, assign this to our global client.
-	if conf.NodeNetSvcAddr != "" {
-		log.G(ctx).Infof("connecting to NodeNetworkService at address %s", conf.NodeNetSvcAddr)
-
-		dialCtx := ctx
-		opts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithStatsHandler(&ocgrpc.ClientHandler{})}
-		if conf.Timeout > 0 {
-			var cancel context.CancelFunc
-			dialCtx, cancel = context.WithTimeout(ctx, time.Duration(conf.Timeout)*time.Second)
-			defer cancel()
-			opts = append(opts, grpc.WithBlock())
-		}
-		client, err := grpc.DialContext(dialCtx, conf.NodeNetSvcAddr, opts...)
-		if err != nil {
-			log.G(ctx).Fatalf("failed to connect to NodeNetworkService at address %s", conf.NodeNetSvcAddr)
-		}
-
-		log.G(ctx).Infof("successfully connected to NodeNetworkService at address %s", conf.NodeNetSvcAddr)
-
-		netSvcClient := nodenetsvc.NewNodeNetworkServiceClient(client)
-		nodeNetSvcClient = &nodeNetSvcConn{
-			addr:     conf.NodeNetSvcAddr,
-			client:   netSvcClient,
-			grpcConn: client,
-		}
-	}
-
-	log.G(ctx).WithFields(logrus.Fields{
-		"TTRPCAddr":      conf.TTRPCAddr,
-		"NodeNetSvcAddr": conf.NodeNetSvcAddr,
-		"GRPCAddr":       conf.GRPCAddr,
-		"Timeout":        conf.Timeout,
-	}).Info("starting ncproxy")
-
-	sigChan := make(chan os.Signal, 1)
-	serveErr := make(chan error, 1)
-	signal.Notify(sigChan, syscall.SIGINT)
-	defer signal.Stop(sigChan)
-
-	// Create new server and then register NetworkConfigProxyServices.
-	server, err := newServer(ctx, conf)
-	if err != nil {
-		log.G(ctx).WithError(err).Fatal("failed to make new ncproxy server")
-	}
-
-	ttrpcListener, grpcListener, err := server.setup(ctx)
-	if err != nil {
-		log.G(ctx).WithError(err).Fatal("failed to setup ncproxy server")
-	}
-
-	server.serve(ctx, ttrpcListener, grpcListener, serveErr)
-
-	// Wait for server error or user cancellation.
-	select {
-	case <-sigChan:
-		log.G(ctx).Info("received interrupt. Closing")
-	case err := <-serveErr:
-		if err != nil {
-			log.G(ctx).WithError(err).Fatal("service failure")
-		}
-	}
-
-	// Cancel inflight requests and shutdown services
-	if err := server.gracefulShutdown(ctx); err != nil {
-		log.G(ctx).WithError(err).Fatal("ncproxy failed to shutdown gracefully")
+	if err := run(); err != nil {
+		logrus.Fatal(err)
```

Comment:
I'd rather us just parse the flags here and call the appropriate functions rather than trying to access the global variables in different places

Reply:
I can parse them in run and pass around, I think I did this run business in case ncproxy might have a second command instead of just it's default use. For instance `ncproxy config` to generate a default config or something. It'll be easier to replace this with `app.Run` in the future

Reply:
Fixed, ptal